proxy/envoy/config_test.sh

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+# Check that envoy can parse golden configuration artifacts
+# In the future, we can run functional tests, but all configs require iptables rules to be effective.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+ENVOY="proxy/envoy/envoy -l trace --service-cluster istio-proxy --service-node 10.0.0.1"
+CONFIG_FILES="envoy-v0 envoy-fault ingress-envoy egress-envoy"
+
+# TODO the test skips over SSL configuration files since referenced cert files are not in the right places
+
+for f in ${CONFIG_FILES}; do
```

Comment:
I think there is some value in doing things the bazel way here. There are many other useful bazel test features we can start incorporating into our tests, but it's hard to do that if we push all the test configuration into test scripts when equivalent features are available in bazel. 

For example, the following options are available at the per-test level with bazel:

- Add the `flaky` attribute and bazel will re-run the test up to three times.
- Add the `manual` tag to exclude tests from wildcard patterns. This is useful if we need to temporarily exclude a test from presubmit without modifying the test code itself. Bonus is the test is still available via `bazel test`.
- The `timeout` attribute can used in this test to avoid polling for SDS/CDC/RDS stats indefinitely and is overridable via `--test_timeout` flag.